### ISSUE TITLE
Base Outline vs Default Outline

### ISSUE THREAD

**Issue by JoomFX** [gantry/gantry5#295]
Guys, why do we have two Outlines? Yeah, most probably I'm missing something but what is the point to have both?

Base is the default ("Global Defaults").
Default is also the default ("Theme Outlines")

It is a bit confusing. Wouldn't it be better to have just the "Base" which will be also a "Default"?

We cannot assign the "Base" to menu items (Assignment tab) but we do not have to as it will be the default and will load to all Pages where we do not have other Outlines assign to.
Also, we CAN assign the "Default" to menu items while we shouldn't. Why would we want to assign the "Default" to a menu item when its purpose is to load on all pages where there is no other Outline assigned?

I'm either very tired or it is indeed confusing :)

**Comment by adi8i** [comment]
I was confused too with this :) . But now for me is something like this: `theme outlines` is where i work,  and `base` is the one that i'll never need to touch and is my third hand, will cover me if i'll miss a color or anything :laughing: .

**Comment by hugoaf** [comment]
Actually the true default template is not specified in Gantry but in Joomla Templates Manager, so we can have 'home' ore any other outline acting as default. Perhaps this issue can be divided in two tasks:

1. a nice notice at top of the Base outline, just explaining what is base outline and something like... the changes made here will be applied to ....etc

2. an icon star or something to identify which of current outlines is acting as true 'default'

**Comment by JoomFX** [comment]
Thank you for your input guys :)

@adi8i You are working for more than a month with G5 and you are quite good with it, so you can find your way without any issues. You also have very good understanding of Joomla so it is not really a problem for you :)
But imaging a user who sees G5 for a very first time. We need to make G5 as user-friendly and intuitive as possible.
By the way, you cannot delete "Hydrogen - Default", I tried it yesterday before opening this issue ;)

@hugoaf You said "*Actually the true default template is not specified in Gantry but in Joomla Templates Manager, so we can have 'home' ore any other outline acting as default.*" - you are completely right, I'm not saying that the default Template/Style is selected from within Gantry. So, how would you select "Base" as default in the Template Manager? You cannot, while you can select "Hydrogen - Default", "Hydrogen - Home" or any custom one you have created.

What I'm trying to achieve with this issue is... to have "Base" and "Default" as one thing. Unless there is something very special behind G5, which I'm not aware of, I do not see the point of having both.

Here's how it should be in my opinion:

![alt](http://i.imgur.com/GNALv4d.png)

And this "Hydrogen - Default (Base)" should be the current "Base" (you should not be able to assign it to a menu item), it should just load on all pages that does not have another "Theme Outline" assigned to :)

**Comment by hugoaf** [comment]
@JoomFX You are absolutely right, a single point for 'Default' with all its meaning will be great. And I think Rockettheme members who post about " I can't see my change in front end" or similar, will be thankful too.

**Comment by JoomFX** [comment]
Thanks @hugoaf !

And a quick correction of what I wrote above... the user should be able to assign "Hydrogen - Default (Base)" to menu items, I mean the "Assignment" tab should be there. The reason is that the user might want to have the Template assigned to some pages and then another template assigned to some other pages. For example:

1. On Homepage the user might want to have Hydrogen
2. On the other pages the user might want to have Callisto

In my opinion the best thing would be to just merge "Base" and "Default" into one Outline. Basically, the same as in G4.

**Comment by mahagr** [comment]
About two months ago, we still had default being the default style in Joomla, but unfortunately it didn't work because of Gantry 5 also specifies content (particle defaults) in the base configuration and changing that base would be fatal to your sites content -- you would loose all your settings when changing the default.

Basically in Joomla there is no master style and no overrides. All the styles are equal, except for a single one that is assigned as fallback - or default - if there's nothing else set. Only a single template can act as a default, so everything else needs to be assigned to a page. There is also no special treatment between the styles. As long as you have more than one style, you can assign or set any of them as your default one. You can also delete any of them as long as you will not delete either the default or the last one.

Here are answers for the questions above:

* You can and should be able to assign menu items to default outline -- it may not be site default
* 'Hydrogen - Default' style can be deleted in Joomla as long as it's not the last one and site default
* Right now G4 has a lot of hacks to work around above (and more) issues in Joomla template manager

Now think of that we would merge base with default.

* As Joomla doesn't make difference between styles, deleting Hydrogen - Default style would be fatal and destroy all your logos etc in your site as other styles were based on it
* Changing default style would also do the same -- even if we copied default settings to the new outline, we could have half of the content changing unexpected as there could have been overrides

Last and not the least, we do not have any control what Joomla does outside our own admin. Styles can be changed from Joomla template manager, where we have no control on. Not to mention that you can change assignments directly from menu items or you can have some 3rd party module manager that replaces Joomla one. So unless we want to rewrite Joomla and all the extensions that do anything with the templates, we really, really need to make our code compatible with the way Joomla has designed those to be used. Otherwise our templates will not work with any 3rd party extension which has something to do with templates.

Basically Base Outline is our global settings that can be overridden in any of the outlines (Joomla styles). As styles are all equal to each other, there will not be Master or Overrides as in Gantry4.

Callisto wasn't the best example on how to use Outlines as the overrides were not working fully, but Base settings will become much more important in the future.

**Comment by mahagr** [comment]
See also #340

**Comment by paulcu** [comment]
I was initially confused by 'base' and 'default' as well but now think of it as similar to how InDesign manages master (templates). I think it's a matter of learning/understanding the terminology. In InDesign you can create a 'base' master (template) and embed it in another 'default' master (templates). You can continue to do this as many times as you like so that you don't have to continue to recreate repeating elements or styles appearing in the 'base' (master) template. Quite powerful if you think about it.

My problem with RC3 is that I can't get some of the configurations in the base template to be applied to the other default and home template. For instance if I configure a copyright and social in the footer area of the 'base' template, it doesn't appear on any of the page views—while the other configurations we set in the base do.

**Comment by JoomFX** [comment]
@paulcu have a look at https://github.com/gantry/gantry5/issues/461

**Comment by wannanah** [comment]
This is so confusing. I am one of the users you're talking about that is confused, so this is a real issue.

**Comment by mahagr** [comment]
It is how Joomla works. There's really no way around it without breaking the whole site if you change the default. The easiest way to think about it is to think that base is common for all the layouts and then default just overrides the common page settings for all languages. In a multi-language site, you would have overrides per every language in the system.
